### DISCUSSION TITLE
Degoogled  Push Notifications

### DISCUSSION THREAD

**Discussion by doggorino** [stoatchat/discussions#656] [Feature Suggestions]
I use a privacy focused version of android called GrapheneOS, it does not come with google services so apps that use firebase for push notifications do not work. If it is possible, I ask the devs to implement push notifications that do not rely on firebase, something like [UnifiedPush](https://unifiedpush.org) for example. Element/FluffyChat (Matrix), Conversations (Xmpp), Fedilab, and Molly/Signal to name a few do not rely  completely on firebase so I am able to get push notifications. Thanks for reading, I hope this gets implemented when you have the time 🙂

**Comment by infi** [comment]
Unfortunately, as our current Android application is a [Trusted Web Activity](https://developer.chrome.com/docs/android/trusted-web-activity), we cannot interface directly with other applications' or system APIs, which includes Unified Push. Push Notifications are handled by your default browser.

**Reply by ghost** [reply]
what about now for the new android app? unifiedpush also has a fallback to FCM, should we open a feature req?

**Reply by IAmTomahawkx** [reply]
We've already written FCM code so another android option isnt really a priority
